state/azure/cosmosdb/cosmosdb.go

```diff
@@ -376,12 +376,23 @@ func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
 	upserts := []CosmosItem{}
 	deletes := []CosmosItem{}
 
+	keyMap := make(map[string]struct{})
+
 	partitionKey := unknownPartitionKey
 
-	for _, o := range request.Operations {
+	// The order of unique key operations does not matter in an atomic transaction.
+	// Only the latest operation for any unique key is selected for execution.
+	// The other operations are redundant, and hence ignored.
+	for i := len(request.Operations) - 1; i >= 0; i-- {
```

Comment:
I think we need to create a new version of the stored procedure where it takes a single array of operations at the same order as the input. This was the main reason I decided to track this as an issue vs putting a small fix in.

As @berndverst  mentioned in the call, the new stored procedure version should be under a new name to avoid conflict with older versions.

Reply:
@artursouza, @berndverst, do we need a stored procedure? I understand that we had some custom logic in place earlier, but now it's just a series of operations. Is this not something that can be directly done from the state store code?